modules/core/src/v2/coins/eth2.ts

```diff
@@ -0,0 +1,767 @@
+  prv: string;
+  recipients: Recipient[];
+}
+
+export interface SignTransactionOptions extends SignFinalOptions {
+  isLastSignature?: boolean;
+  expireTime: number;
+  sequenceId: number;
+  gasLimit: number;
+  gasPrice: number;
+}
+
+export interface HalfSignedTransaction extends BaseHalfSignedTransaction {
+  halfSigned: {
+    recipients: Recipient[];
+    expireTime: number;
+    contractSequenceId: number;
+    sequenceId: number;
+    txHex?: never;
+  };
+}
+
+export interface FullySignedTransaction {
+  txHex: string;
+}
+
+export type SignedTransaction = HalfSignedTransaction | FullySignedTransaction;
+
+interface PrecreateBitGoOptions {
+  enterprise?: string;
+  newFeeAddress?: string;
+}
+
+interface OfflineVaultTxInfo {
+  nextContractSequenceId?: string;
+  contractSequenceId?: string;
+  tx: string;
+  userKey: string;
+  backupKey: string;
+  coin: string;
+  gasPrice: number;
+  gasLimit: number;
+  recipients: Recipient[];
+  walletContractAddress: string;
+  amount: string;
+  backupKeyNonce: number;
+}
+
+interface UnformattedTxInfo {
+  recipient: Recipient;
+}
+
+export interface RecoverOptions {
+  userKey: string;
+  backupKey: string;
+  walletPassphrase?: string;
+  walletContractAddress: string;
+  recoveryDestination: string;
+}
+
+export interface RecoveryInfo {
+  id: string;
+  tx: string;
+  backupKey?: string;
+  coin?: string;
+}
+
+interface GetSendMethodArgsOptions {
+  recipient: Recipient;
+  expireTime: number;
+  contractSequenceId: number;
+  signature: string;
+}
+
+interface SendMethodArgs {
+  name: string;
+  type: string;
+  value: any;
+}
+
+interface FeeEstimate {
+  gasLimitEstimate: number;
+  feeEstimate: number;
+}
+
+export class Eth2 extends BaseCoin {
+  static createInstance(bitgo: BitGo): BaseCoin {
+    return new Eth2(bitgo);
+  }
+
+  /**
+   * Returns the factor between the base unit and its smallest subdivison
+   * @return {number}
+   */
+  getBaseFactor(): string {
+    // 10^18
+    return '1000000000000000000';
+  }
+
+  getChain(): string {
+    return 'eth2';
+  }
+
+  getFamily(): string {
+    return 'eth2';
+  }
+
+  getFullName(): string {
+    return 'Ethereum 2.0';
+  }
+
+  /**
+   * Flag for sending value of 0
+   * @returns {boolean} True if okay to send 0 value, false otherwise
+   */
+  valuelessTransferAllowed() {
+    return true;
+  }
+
+  /**
+   * Flag for sending data along with transactions
+   * @returns {boolean} True if okay to send tx data (ETH), false otherwise
+   */
+  transactionDataAllowed() {
+    return true;
+  }
+
+  /**
+   * Evaluates whether a address string is valid for this coin
+   * @param address
+   */
+  isValidAddress(address: string): boolean {
+    return this.isValidPub(address);
+  }
+
+  /**
+   * Return boolean indicating whether input is valid public key for the coin.
+   *
+   * @param {String} pub the pub to be checked
+   * @returns {Boolean} is it valid?
+   */
+  isValidPub(pub: string): boolean {
```

Comment:
didn't you impl this in account-lib? can just reuse

Reply:
The check was for `isValidPriv` --> looking at it it's simpler to keep it in here I think, allows us to avoid initializing a keypair() object before checking if the pubkey is valid